kratos/linear_solvers/amgcl_solver.h

```diff
@@ -489,9 +489,9 @@ class AMGCLSolver : public LinearSolver< TSparseSpaceType,
      * @brief This method returns the current iteration number
      * @return mIterationsNumber The current iteration number
      */
-    virtual IndexType GetIterationsNumber()
+    virtual unsigned int GetIterationsNumber() override
     {
-        return mIterationsNumber;
+        return static_cast<unsigned int>( mIterationsNumber );
```

Comment:
Maybe it's better to add the `IndexType` to the base class to avoid the cast.

Reply:
Actually there is TUblasSparseSpace<double>::IndexType accessible, and for consistency, all instances of iteration numbers should be of that type (actually is the same as AMGCL::IndexType). However, mIterationsNumber is of "unsigned int" type in iterative_solver and "IndexType" for AMGCL. Anyway, I can change the type for mIterationsNumber.

Reply:
> Anyway, I can change the type for mIterationsNumber.

:+1:  I'd do so.

Reply:
I further checked the IterativeSolver (and derived classes), there are several instances of comparing mIterationsNumber with mMaxIterationsNumber, which is an "unsigned int". Therefore, I think it is more efficient to have this static cast here to prevent those inherent run-time conversions (whenever IndexType is needed to be compared with an unsigned int).

Reply:
@rubenzorrilla could you please hint on this issue?

Reply:
I'd change all to have a consistent `IndexType` in both the `mIterationsNumber` and the `mMaxIterationsNumber`. By doing this I think we can avoid the cast(s).

Reply:
@rubenzorrilla please check.